src/rgw/rgw_rest_swift.cc

```diff
@@ -1803,11 +1803,10 @@ bool RGWSwiftWebsiteHandler::is_web_dir() const
   std::string subdir_name;
   url_decode(s->object.name, subdir_name);
 
-  /* Remove character from the subdir name if it is "/". */
   if (subdir_name.empty()) {
     return false;
   } else if (subdir_name.back() == '/') {
-    subdir_name.pop_back();
+    return true;
```

Comment:
It looks a little strange. In my test, without this patch the index.html in sub-dir test1 cannot be shown and it returned 404. Only the index.html in container web1 can be shown.
```
[root@CephS debugfile]# curl -i "${publicURL}/web1/" -X GET
HTTP/1.1 200 OK
Content-Length: 22
Accept-Ranges: bytes
Last-Modified: Wed, 05 Apr 2017 12:47:27 GMT
X-Timestamp: 1491396447.03588
etag: e67dde62508cb6c715eba5fae0f85646
X-Object-Meta-Mtime: 1491396423.695949
X-Trans-Id: tx000000000000000000040-0058e4e767-1018-default
Content-Type: binary/octet-stream
Date: Wed, 05 Apr 2017 12:47:35 GMT

This is a index.html!
[root@CephS debugfile]# 
[root@CephS debugfile]# curl -i "${publicURL}/web1/test1" -X GET
HTTP/1.1 404 Not Found
Content-Length: 9
X-Trans-Id: tx000000000000000000041-0058e4e780-1018-default
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Date: Wed, 05 Apr 2017 12:48:00 GMT

NoSuchKey[root@CephS debugfile]# curl -i "${publicURL}/web1/test1/" -X GET
HTTP/1.1 404 Not Found
Content-Length: 9
X-Trans-Id: tx000000000000000000042-0058e4e784-1018-default
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Date: Wed, 05 Apr 2017 12:48:04 GMT

NoSuchKey
```

but the index.html in sub-dir is exactly exitent
```
[root@CephS debugfile]# swift list web1
index.html
test1/index.html
```